### PR TITLE
Episode 1

### PR DESCRIPTION
**Pull request by brendan-kellam**
99% of these are unity assets / source Brackey provided.

The files that are actual changes are anything located in the "Scripts" folder + "Transcripts" folder. 

I figured we could use "transcripts" to get a look at what was done on the editor side (I.E things we can't review). They would also be valuable when we actually have to present... Personally, I know I'm going to forget half of the configuration steps within 5 minutes lol

Also, the repo was auto-configured with git large file storage. I've never used it tbh, but it seems to just kinda work along side git without any hassle. Download: https://git-lfs.github.com/

### CODE REVIEW

File: Transcripts/Episode1.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+Talk aout pivot vs. center && global vs. local
+
+Add Components to player:
+
+- Nav mesh agent
+    - Adjust speed, accel and angular accel
+- Capsule collider
+    - Turn off gfx to see, move up collider to be contained within the nav mesh agent
+- Rigid body
```
Comment (Vassenx) [inline Transcripts/Episode1.md:16]: Go over rigidbody, transform, kinematic, and nav mesh agent all together here?

File: Transcripts/Episode1.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# Episode 1
+### [Youtube Link](https://www.youtube.com/watch?v=S2mK6KFdv0I&list=PLPV2KyIb3jR4KLGCCAciWQ5qHudKtYeP7)
+### [Notion](https://www.notion.so/gamedevmcgill/Dissection-of-Brackey-s-RPG-25c5b38888d840a5b5da528644c5a9ea#7c5b0616c1d942789e4f71fc1fbb1712)
+
+
```
Comment (Vassenx) [inline Transcripts/Episode1.md:5]: **unity setup** (<5 mins)
Review what is a scene, game object, component
Quickly show the scene vs game view, hierarchy window, folder setup

File: Transcripts/Episode1.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+**Player Motor**
+- nothing much here
+
+**Camera**
+
+- Introduces LateUpdate here
+- Also transform.LookAt
+- First time he uses a public transform field in a component to reference another gameobject (The player)
+- Sets up camera position at runtime and applies this to the prefab by copying the values over.
+- He multiplies yawSpeed by Time.deltaTime. This would be worth to explain.
```
Comment (Vassenx) [inline Transcripts/Episode1.md:78]: also explain what yaw is
explain Time.time / Time.fixedTime along with Time.deltaTime

File: Transcripts/Episode1.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+- Navigation > Bake
+    - Agent Radius → 0.4
+    - Step Height → 0.5
+
+**Player Controller**
+
+- Add new script called PlayerController
+- TODO: May need to explain what the "out" keyword does. Passing by value / ref
+- Simple raycast system with left-click.
```
Comment (Vassenx) [inline Transcripts/Episode1.md:62]: Explain raycast

File: GDM Bootcamp RPG/Assets/Scripts/PlayerController.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+        if (Input.GetMouseButtonDown(0))
+        {
+            Ray ray = cam.ScreenPointToRay(Input.mousePosition);
+            RaycastHit hit;
+
+            // TODO: May need to explain what the "out" keyword does. Passing by value / ref
+            if (Physics.Raycast(ray, out hit, 100, movementMask))
+            {
+                motor.MoveToPoint(hit.point);
```
Comment (Vassenx) [inline GDM Bootcamp RPG/Assets/Scripts/PlayerController.cs:33]: Include WASD controls
Do we want to show rigidbody w/ AddForce or velocity real fast?

File: Episode1_Explanations.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
+# Episode 1
+
+* [Scene, Gameobject, and Components](#scene,-gameobject,-and-components)
+* [The Game Loop](#the-game-loop)
+* [Ray Casting](#ray-casting)
+
+## Scene, Gameobject, and Components
```
Comment (msukkari) [inline Episode1_Explanations.md:7]: Scene, Gamobject, and Components explanation

File: Episode1_Explanations.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+### Components
+A Component in Unity is an object which may contain both data and logic, that can be added to a Gameobject. Components are where the actual behaviour and appearence of a Gameobject are implemented. Therefore, by attaching different types of Components to a specific Gameobject, the behaviour and appearence of the Gameobject is created. For example, Unity's [RigidBody](https://docs.unity3d.com/ScriptReference/Rigidbody.html) component, when attached to a Gameobject, allows the Gameobject to be effected by Unity's physics system (i.e. collide with objects, experience gravity, etc). Unity provides several other built-in Components, and also allows you to create your own Components using C# or Javascript scripts.\
+#### Further reading
+https://docs.unity3d.com/510/Documentation/Manual/GameObjects.html \
+https://docs.unity3d.com/510/Documentation/Manual/TheGameObject-ComponentRelationship.html
+
+## The Game Loop
```
Comment (msukkari) [inline Episode1_Explanations.md:19]: Game Loop explanation

File: Episode1_Explanations.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+	for(Component* component : components) 
+	{
+		component->Update();
+	}
+}
+```
+We have only started to scratch the surface on what a game loop is, and how we can interact with it in Unity. Some extremely valuable resources which I recommend reading are:\
+[https://docs.unity3d.com/Manual/ExecutionOrder.html](https://docs.unity3d.com/Manual/ExecutionOrder.html): Unity's discription of the execution order of functions in it's game loop (especially the ones that you can override, such as Update and FixedUpdate)\
+[https://gameprogrammingpatterns.com/game-loop.html](https://gameprogrammingpatterns.com/game-loop.html): An amazing explanation of a game loop and some common pitfalls associated with it. 
+## Ray Casting
```
Comment (msukkari) [inline Episode1_Explanations.md:45]: Ray Casting explanation

File: Episode1_Explanations.md (hunk 7 above)
Comment (Vassenx) [reply Episode1_Explanations.md:19]: The c++ code may be too confusing for beginners. Instead maybe c# pseudo
List<Components> components = GetAllComponents();
for(int i = 0; i < components.Count; i++){
    components[i].Update();
}

File: Episode1_Explanations.md [hunk 9]
```diff
@@ -53,21 +53,29 @@ https://www.scratchapixel.com/lessons/3d-basic-rendering/minimal-ray-tracer-rend
...
+While CharacterController gives you more control over what movements you want, Rigidbody code is a little easier to understand because a lot of the movement is using the AddForce function. \
+For a complete explanation with code: https://medium.com/ironequal/unity-character-controller-vs-rigidbody-a1e243591483. \
+As mentioned in the link, if you are beginner and struggling with moving around slopes and stairs, a CharacterController does this automatically. Rigidbody does not let objects go up stairs without extra work. Think of a ball hitting the bottom of a staircase, it won't just go up the stairs on its own. Raycasting can help with this by raycasting to the ground in front of you to see if there is ground in the way, but there are quite a few tricks to achieve this. \
+Below is a simple walk Rigidbody setup, which checks for axis input (WASD, arrows, etc) https://docs.unity3d.com/ScriptReference/Input.GetAxis.html and uses MovePosition. You can set the Rigidbody.position instead of MovePosition to teleport instead of smoothly moving to a location.\
+
 ```
```
Comment (Vassenx) [inline Episode1_Explanations.md:63]: Rigidbody vs CharacterController

**Review by Vassenx — CHANGES_REQUESTED** [review]

**Pull request merged**